ladybug/_datacollectionbase.py

```diff
@@ -623,3 +681,42 @@ def __repr__(self):
         """Discontinuous Collection representation."""
         return "Discontinuous Data Collection\n{} ({})\n...{} values...".format(
             self.header.data_type, self.header.unit, len(self._values))
+
+
+class _DataCollectionEnumeration(object):
```

Comment:
This is nice to have but can't we just hard-code these classes? unlike data-types we won't have so many datacollections that might be added to ladybug.

Reply:
The difficulty there is that you will get circular imports of the data collections that inherit from one another.  Take, for example a standard mutable hourly collection. When you import this collection from datacollection.py, it does not import its immutable counterpart because its immutable counterpart inherits from the mutable one.  If we want to have a `to_immutable()` method on this object, there needs to be some way of importing the immutable one outside of the object itself.  I guess we could do this by copying a good fraction of this code that automatically imports all data collections into `__init__.py` if that sounds ok to you.

Then we could move the code that searches through all of the child classes of BaseCollection onto the BaseCollection object for all of the other classes to use.

Let me know if this sounds like the right solution to you and I can implement it.

Reply:
Thanks. Makes sense. I think the current implementation is good.  Let's keep it as it is.